src/troubleshooting/errors.rst

```diff
@@ -0,0 +1,118 @@
+SSSD ERRORS
+##########
+
+
+SSSD process is terminated by own WATCHDOG
+******************************************
+
+.. code-block:: bash
+
+  (Fri Apr 14 15:07:19 2023) system1 sssd[sssd]: Child [1277] ('SSSDdomain':'%BE_SSSDdomain') was terminated by own WATCHDOG. Consult corresponding logs to figure out the reason.
+
+The above log indicates that the sssd_be process was blocked too long on something that was longer than ``3*10`` seconds. In order to find which operation is blocking sssd_be, enable sssd debugging by adding ``debug_level = 9`` under all sections of the ``/etc/sssd/sssd.conf`` file, especially under the ``[$domain]`` section, and wait for the issue to reoccur. Once the issue is observed, take the timestamp of the ``... was terminated by own WATCHDOG`` message and then spot the last operation before the timestamp in ``/var/log/sssd/sssd_$domain.log``. In the above example, sssd_be is getting killed, but there could be some other situation where ``sssd_nss`` or ``sssd_pam`` processes could get killed by watchdog. In this case, the troubleshooting process will be the same, but the respective sssd log needs to be checked instead of the ``/var/log/sssd/sssd_$domain.log`` file. Increasing the ``timeout`` values may server as a workaround. But its always good to spot the operation that causing the issue(sss process terminated by WD) and apply the solution accordingly. On a side note if the processing of group membership is very slow then also we can observe ``sssd_be killed by watchdog``.
+
+  
+sdap_async_sys_connect request failed
+*************************************
+
+.. code-block:: bash
+
+  (Fri Apr 14 16:07:19 2023) [sssd[be[example.com]]] [sssd_async_socket_init_done] (0x0020): sdap_async_sys_connect request failed: [110]: Connection timed out.
+
+sdap_async_sys_connect request failed is being noticed if sssd is not able to connect to the LDAP server within 6 seconds. This could be an issue with DNS or the network. Validate the DNS SRV records; if SRV records are not working, hardcoding the AD/LDAP server may help here Example: If id_provider = ad is being used, then hardcoding of AD servers can be done as: add ``ad_server = ad1.example.com, ad2.example.com`` under the ``[$domain]`` section of the ``/etc/sssd/sssd.conf``. But if the network is slow or ldap_network_timeout is timed out, then consider increasing the value of ldap_network_timeout, which is set to 6 seconds by default.
```

Comment:
`Example: If id_provider = ad is being used, then `

Would change this to:

For example, if `id_provider = ad` is being used then....